test/support/factory.ex

```diff
@@ -335,4 +335,25 @@ defmodule Skate.Factory do
       ]
     }
   end
+
+  def ueberauth_auth_factory do
+    %Ueberauth.Auth{
+      provider: :keycloak,
+      uid: "test_username",
+      credentials: %Ueberauth.Auth.Credentials{
+        expires_at: System.system_time(:second) + 1_000,
+        refresh_token: "test_refresh_token"
+      },
+      info: %{email: "test@mbta.com"},
+      extra: %Ueberauth.Auth.Extra{
+        raw_info: %UeberauthOidcc.RawInfo{
+          userinfo: %{
+            "resource_access" => %{
+              "test-client" => %{"roles" => ["test1", "skate-readonly"]}
+            }
+          }
+        }
+      }
+    }
+  end
```

Comment:
I took the lazy route and just pulled one of the working auth structs from `auth_controller_test` but I'm happy to make this factory "better". I figured we could punt on it till later though

Reply:
Yeah I'm fine with the "get it working and then improve it when we need to" approach on this one.